### DISCUSSION TITLE
Kafka2Rest - Manage REST response

### DISCUSSION THREAD

**Discussion by vguccione01** [mia-platform/community#345] [Q&A] [answered]
### Which section?

Console

### Which is the version?

1.1.0

### Description

Goodmorning, 
I'm currently working on a project where we are using the microservice Kafka2Rest to intercept a command from Flow Manager to perform a REST call to another microservice.

My question is:
It is possible through Kafka2Rest manage the response of the REST call? I noticed that the microservice already log the response.

Thanks in advance!

**Comment by lucascanna** [comment, marked as answer]
Hi @vguccione01,

Thank you for reaching out! Currently, Kafka2Rest doesn't provide any customization options for handling the HTTP response directly. However, the microservice does log the response, which you've already observed.

To better assist you, could you please provide more details on what specific customization or management of the HTTP response you have in mind? Understanding your requirements will help us explore possible solutions or consider improvements for future releases.

Looking forward to your insights!

**Reply by vguccione01** [reply]
Hi @Lucascanna 
I would like to submit a message on the kafka topic based on some values of the HTTP response body.
Would be great to have a response handler configuration like the ones for `validatorProcessor`, `pathProcessor` and `bodyProcessor`

**Reply by lucascanna** [reply]
Hello,

Thank you for clarifying your requirements. While I understand the need to submit a message on the Kafka topic based on the  HTTP response values, it's important to note that Rest2Kafka is designed for simpler use cases involving straightforward HTTP calls.

For more complex scenarios like the one you've described, I would recommend considering the implementation of a custom Kafka consumer. This approach provides greater flexibility and control over handling the HTTP response, allowing you to tailor the solution to your specific needs.

Introducing a response handler configuration akin to validatorProcessor, pathProcessor, and bodyProcessor could indeed add complexity to the service. For example, it would involve exposing libraries for making HTTP calls within the response handler.

**Reply by vguccione01** [reply]
Hi,
I understand the complexity problem, so in more complex scenarios is better to implement a custom service.
Thank you for the assistance!
